### ISSUE TITLE
--check does not check syntax errors

### ISSUE THREAD

**Issue by BhaagBoseDK** [accumulator/charge-lnd#41]
--check does not check syntax error.

for 
[new section]
node.id = something

I had a - instead of _ mistake in 
max.chan-ratio (instead of max.chan_ratio

Also I had spelling mistake

strategy = proportional

These were not caught during --check and instead they were caught in --dry-run

**Comment by accumulator** [comment]
You are correct, many parts of the config file are evaluated late, after --check does its thing. --check basically only does a 'does this look like an INI style file?'

**Comment by BhaagBoseDK** [comment]
may be worth checking syntax as well as part of --check.

**Comment by accumulator** [comment]
You are correct again, that's why I kept this bug report open :)
